ProteinCartography/fetch_accession.py

```diff
@@ -67,8 +64,9 @@ def fetch_pdb(accession: str, output_dir: str):
     source = "https://alphafold.ebi.ac.uk/files/AF-{}-F1-model_v4.pdb".format(accession)
 
     if not os.path.exists(output_path):
+        user_agent = USER_AGENT_HEADER["User-Agent"]
```

Comment:
Do we think it would be possible to replace this `curl` call with `requests`?

Reply:
Yes, I'll take a look at including that call as well into this PR today.

Reply:
It looks like currently with the curl code a ".pdb" file is created even when an API call fails and instead of having structure information it just has the request error message. Many of them are 404s. When I change this behavior it leads to a failure of the pipeline as those files are no longer created. It's possible to do something similar to before and create place holder/empty files, but that seems odd. Is there a particular behavior you are looking for? Depending I might make this a separate PR if it involves more extensive changes.

Reply:
I think the crashing stems from the way that Snakemake touches and handles files and looks for missing files to calculate the DAG.

1. after BLAST and Foldseek, we aggregate all hits and filter them to get a final list of PDBs we'd like to download
2. for each protein in this list, we create a dummy file via the Snakemake checkpoint `create_alphafold_wildcard`
3. this spawns an empty file with name `protid.pdb` in the `alphafold_dummy` directory
4. from my understanding, this is needed for Snakemake to be able to create a new job for each protein in the `download_pdbs` rule
5. Snakemake will fail if a PDB file is not created for each dummy file, so it's easier just to create a PDB with the error message inside
6. These errored proteins don't get added to the Foldseek local clustering analysis, and also don't get analyzed by `assess_pdbs.py` because it looks for part of the error message and ignores protein with `Error` in the contents

Ideally, we'd have a way to do this that avoids creating the dummy file if the protein doesn't have an AlphaFold structure. This could be possible if we filter the UniProt hits by looking to see whether they have an AlphaFold entry, but in my experience these aren't always accurate – the UniProt page sometime says that there isn't an AlphaFold structure even when there is (I think I've also seen the case vice-versa).

I think it would also be nice to avoid needing to create the dummy file altogether, but it seems like this is the most straightforward way to allow for re-evaluation of the DAG mid-pipeline. From my understanding this behavior would be obviated if we used Nextflow instead, but that's a much bigger can of worms.

Reply:
Oh ok. Well then I'll just write the same error message to the file and make sure it still works that way it's doing the same thing. I just wanted to make sure I wasn't going to propagate something that shouldn't be happening, but that makes sense.

Reply:
That sounds fine for now! This overall behavior is definitely not ideal, especially if people are going to be opening those PDB files expecting actual content. we should figure out a smarter way to deal with this more sensibly in the future